### PR TITLE
changelog: update changelog

### PR DESCRIPTION
**Pull request by sam-golioth**
Update changelog with all changes since v0.9.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,15 +4,67 @@ All notable changes to this project will be documented in this file.
...
+  for each service that they use.
+- Stream and State: change `json` and `cbor` object set calls to specify
+  type in the parameters (instead of separate function calls).
+- RPC: This service must now be initialized prior to use by calling
+  `golioth_rpc_init(client);`
+- Settings: This service must now be initialzied prior to use by calling
+  `golioth_settings_init(client);`
+- Services (e.g. stream, fw_update, etc.) are disabled by default - be
+  sure to enable them in your project configuration.
+- Typedefs have been removed through the codebase - instead use the
```
Comment (szczys) [inline CHANGELOG.md:27]:
```suggestion
- Typedefs have been removed throughout the codebase - instead use the
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,15 +4,67 @@ All notable changes to this project will be documented in this file.
...
+  underlying struct or enum types
```
Comment (szczys) [inline CHANGELOG.md:28]:
```suggestion
  underlying struct or enum types
- Remove `lightdb_` from `lightdb_stream` functions to differentiate from LIghtDB State
```

**Review by szczys — APPROVED** [review]
Looks good, two suggestions that aren't must-haves.

File: CHANGELOG.md (hunk 2 above)
Comment (sam-golioth) [reply CHANGELOG.md:28]: Hmm, I thought I added that. Good catch!

**Pull request merged**